lib/src/components/drawer/DrawerContent.tsx

```diff
@@ -0,0 +1,109 @@
+import { Content, Portal } from '@radix-ui/react-dialog'
+import React from 'react'
+
+import { DIALOG_Z_INDEX } from '~/constants/zIndices'
+import { styled } from '~/stitches'
+import { slideInOut } from '~/utilities'
+
+import { DrawerOverlay } from './DrawerOverlay'
+import { DrawerContext } from './Drawer.context'
+import { Flex } from '~/components/flex'
+
+import { ColorScheme } from '~/experiments/color-scheme'
+
+const animationTop = slideInOut({ outPosition: { x: 0, y: '-100%' } })
+const animationRight = slideInOut({ outPosition: { x: '100%', y: 0 } })
+const animationBottom = slideInOut({ outPosition: { x: 0, y: '100%' } })
+const animationLeft = slideInOut({ outPosition: { x: '-100%', y: 0 } })
+const setupAnimation = (animation) => {
```

Comment:
You could make it a bit more typesafe:
```suggestion
const setupAnimation = (animation: ReturnType<typeof slideInOut>) => {
```